src/pages/workspaces/workspace/analysis/runtime-utils.ts

```diff
@@ -0,0 +1,142 @@
+  const validGpuOptionsForZone = getValidGpuTypesForZone(zone)
+  const validGpuOptions = _.filter(({ maxNumCpus, maxMem, type }) => numCpus <= maxNumCpus && mem <= maxMem && validGpuOptionsForZone.includes(type),
+    gpuTypes)
+  return validGpuOptions || { name: '?', type: '?', numGpus: '?', maxNumCpus: '?', maxMem: '?', price: NaN, preemptiblePrice: NaN }
+}
+
+export const trimRuntimesOldestFirst = (runtimes: Runtime[]): Runtime[] => {
+  const runtimesWithoutDeleting: Runtime[] = _.remove({ status: 'Deleting' }, runtimes)
+  const sortedRuntimes: Runtime[] = _.sortBy('auditInfo.createdDate', runtimesWithoutDeleting)
+  return sortedRuntimes
+}
+
+// Status note: undefined means still loading and no runtime
+export const getCurrentRuntime = (runtimes: Runtime[]): Runtime | undefined => !runtimes ? undefined : (_.flow(trimRuntimesOldestFirst, _.last)(runtimes) || undefined)
+
+// TODO: This function needs to be removed once we have testing in place in all places its used
+// DO NOT USE THIS
+export const getRuntimeForTool = (toolLabel: ToolLabel, currentRuntime: Runtime, currentRuntimeTool: RuntimeToolLabel): Runtime | undefined => Utils.cond([toolLabel === currentRuntimeTool, () => currentRuntime],
+  [Utils.DEFAULT, () => undefined])
+
+export const getAnalysesDisplayList = _.flow(
+  _.map(
+    _.flow(
+      _.get('name'),
+      _.split('/'),
+      _.nth(1)
+    )
+  ),
+  _.without([undefined]),
+  _.join(', ')
+)
+
+
+export const getConvertedRuntimeStatus = (runtime: Runtime): string => {
+  return runtime && (runtime.patchInProgress ? 'LeoReconfiguring' : runtime.status) // NOTE: preserves null vs undefined
+}
+
+export const getComputeStatusForDisplay = (status: LeoRuntimeStatus): DisplayRuntimeStatus => Utils.switchCase(_.lowerCase(status),
+  ['leoreconfiguring', () => 'Updating'],
+  ['starting', () => 'Resuming'],
+  ['stopping', () => 'Pausing'],
+  ['stopped', () => 'Paused'],
+  ['prestarting', () => 'Resuming'],
+  ['prestopping', () => 'Pausing'],
+  [Utils.DEFAULT, () => _.capitalize(status)])
+
+export const displayNameForGpuType = (type: string): string => {
+  return Utils.switchCase(type,
+    ['nvidia-tesla-k80', () => 'NVIDIA Tesla K80'],
+    ['nvidia-tesla-p4', () => 'NVIDIA Tesla P4'],
+    ['nvidia-tesla-v100', () => 'NVIDIA Tesla V100'],
+    ['nvidia-tesla-p100', () => 'NVIDIA Tesla P100'],
+    [Utils.DEFAULT, () => 'NVIDIA Tesla T4']
+  )
+}
+
+export const getIsRuntimeBusy = (runtime: Runtime): boolean => {
+  const { Creating: creating, Updating: updating, LeoReconfiguring: reconfiguring, Stopping: stopping, Starting: starting } = _.countBy(getConvertedRuntimeStatus, [runtime])
+  return _.sum([creating, updating, reconfiguring, stopping, starting]) > 0
+}
+
+// NOTE: the label property is being compared to Ajax response values, so the label cannot be changed without
+// impacting code.
+export const cloudProviders: { [label: string]: { label: CloudProvider } } = {
```

Comment:
Also ideally renamed to CloudPlatform

Reply:
Once again, do not think that conforming to new standards of consistency unrelated to runtimes should be within the scope of this PR, seems like a prime candidate for a small one-off PR since the renames will touch a lot of code